src/test/ui/type-alias-impl-trait/bound_reduction2.stderr

```diff
@@ -1,5 +1,14 @@
+error[E0277]: the trait bound `T: TraitWithAssoc` is not satisfied
```

Comment:
Huh, this error is definitely not great. Do you understand what's going on here? (I don't, at least not yet)

Reply:
I think this is because `T` ends up in the context of the `impl Trait` which doesn't have the parameter nor the relevant bound in its `ParamEnv`.
It's not great but this is a pre-existing bug in the existential type aliases feature that was being hidden because the `Reveal` mode was getting ignored after the first time.